### PR TITLE
YAK-2340/removed template flag

### PR DESCRIPTION
**Pull request by samadsid7**

### CODE REVIEW

File: operator/controllers/utils.go [hunk 1]
```diff
@@ -464,9 +459,9 @@ func validateConfiguration(ctx context.Context, log logr.Logger, namespace strin
 
 func buildStatefulSet(name string, namespace string, baseImage string, isBootstrap bool,
 	configuration kvstorev1.HbaseClusterConfiguration, configVersion string, fsgroup int64,
-	d kvstorev1.HbaseClusterDeployment, log logr.Logger, statefulSetLabelsUpdateValue string) *appsv1.StatefulSet {
+	d kvstorev1.HbaseClusterDeployment, log logr.Logger, isHbaseCluster bool) *appsv1.StatefulSet {
```
Comment (rda3mon) [inline operator/controllers/utils.go:462]: what is the intention of this variable? `isHbaseCluster?` correctly named?

File: operator/controllers/utils.go [hunk 2]
```diff
@@ -849,9 +839,8 @@ func labelsForPodService(crName string, name string, labels map[string]string) m
 	}
 }
 
-// labelsForHbaseCluster returns the labels for selecting the resources
-// belonging to the given hbasecluster CR name.
-func labelsForHbaseCluster(name string, labels map[string]string) map[string]string {
+// common template and match labels for cluster
+func generateCommonLabels(name string, labels map[string]string) map[string]string {
```
Comment (rda3mon) [inline operator/controllers/utils.go:843]: can you use better naming or more appropriate naming?

File: operator/controllers/utils.go [hunk 3]
```diff
@@ -487,19 +482,14 @@ func buildStatefulSet(name string, namespace string, baseImage string, isBootstr
...
-
-	//Assign selectorMatchLabelsMap existing labels . If statefulSetLabelsUpdateValue set to matchLabels then update the selectorMatchLabelsMap
 	selectorMatchLabelsMap := ls
-	if statefulSetLabelsUpdateValue == "matchLabels" {
-		templateLabelsMap = updateTemplateLabelsForStatefulSet(name, d.Name, templateLabelsMap)
-		selectorMatchLabelsMap = updateMatchLabelsForStatefulSet(name, d.Name)
-		log.Info("Deploying StatefulSet with new selector match Labels for", "StatefulSet:", d.Name)
+
+	// If its hbase core cluster , assign templateLabels and matchLabels different labels on top of existing labels
+	if isHbaseCluster {
```
Comment (rda3mon) [inline operator/controllers/utils.go:490]: better flag would be ismultistatefulset?

**Review by rda3mon — CHANGES_REQUESTED** [review]

File: operator/controllers/utils.go [hunk 4]
```diff
@@ -585,7 +578,7 @@ func buildService(svcName string, crName string, namespace string, labels map[st
 			Type:                     corev1.ServiceTypeClusterIP,
 			ClusterIP:                "None",
 			PublishNotReadyAddresses: true,
-			Selector:                 labelsForHbaseCluster(svcName, selectorLabels),
+			Selector:                 createSharedLabelsMap(svcName, selectorLabels),
```
Comment (rda3mon) [inline operator/controllers/utils.go:581]: create or get?

**Review by rda3mon — CHANGES_REQUESTED** [review]

**Review by rda3mon — APPROVED** [review]

**Review by manishankarbalu — APPROVED** [review]

**Pull request merged**